Simplenote/Classes/UITextView+Simplenote.swift

```diff
@@ -34,3 +60,56 @@ extension UITextView {
         return attributedText.attribute(.attachment, at: index, effectiveRange: nil) as? T
     }
 }
+
+
+// MARK: - Geometry
+//
+extension UITextView {
+
+    /// Returns the Bounding Rect for the specified `Range<String.Index>`
+    ///
+    func boundingRect(for range: Range<String.Index>) -> CGRect {
+        let nsRange = text.utf16NSRange(from: range)
+        return boundingRect(for: nsRange)
+    }
+
+    /// Returns the Bounding Rect for the specified NSRange
+    ///
+    func boundingRect(for range: NSRange) -> CGRect {
+        let glyphRange = layoutManager.glyphRange(forCharacterRange: range, actualCharacterRange: nil)
+        let rect = layoutManager.boundingRect(forGlyphRange: glyphRange, in: textContainer)
+
+        return rect.offsetBy(dx: textContainerInset.left, dy: textContainerInset.top)
+    }
+
+    /// Returns the "Editing Rect": We rely on this calculation to determine the "available area" in which Interlinks Autocomplete
+    /// can be presented.
+    ///
+    /// - Note: `contentInset.bottom` is expected to contain the bottom padding required by the keyboard. Capisce?
+    ///
+    func editingRect() -> CGRect {
+        let paddingTop = safeAreaInsets.top
+        let paddingBottom = safeAreaInsets.bottom + contentInset.bottom
+        let editingHeight = frame.height - paddingTop - paddingBottom
+
+        return CGRect(x: .zero, y: paddingTop, width: frame.width, height: editingHeight)
+    }
+
+    /// Returns the Window Location for the text at the specified range
+    ///
+    func locationInSuperviewForText(in range: Range<String.Index>) -> CGRect {
+        let rectInEditor = boundingRect(for: range)
+        return superview?.convert(rectInEditor, from: self) ?? rectInEditor
```

Comment:
To convert this rect to window coordinates you can do: 
```
convert(rectInEditor, to: nil)
```
I think it's safer to convert it to window coordinates here and let the receiver to convert it from window coordinates to local coordinates.